esmvalcore/cmor/check.py

```diff
@@ -464,17 +464,50 @@ def _check_generic_level_dim_names(self, key, coordinate):
             else:
                 if out_name:
                     self.report_critical(
-                        f'Generic level coordinate {key} '
-                        'has wrong standard_name '
-                        'or is not set.', )
+                        f'Generic level coordinate {key} with out_name '
+                        f'{out_name} has wrong standard_name or is not set.')
                 else:
-                    self.report_critical(self._does_msg, key, 'exist')
+                    self._check_alternative_dim_names(key)
+
+    def _check_alternative_dim_names(self, key):
+        """Check for viable alternatives to generic level coordinates."""
+        alternative_coord = None
+        allowed_alternatives = {
+            'alevel': ['alt40', 'plev19', 'plevs'],
```

Comment:
I think it would be helpful to indicate (probably in a comment) where this information is coming from. It seems to somehow come from CMIP6 tables, but then, there is no mention of `plevs` in the tables, and on the other hand one might wonder why not mention `alt16` or the other `plevXX`.

Reply:
Yes, very good point! I will add an explanation of that. `plevs` comes from the CMIP5 table, btw.